### PR TITLE
Fix/test generators

### PR DESCRIPTION
**Pull request by denis-fesechko**

### CODE REVIEW

File: generators/auth/templates/mongodb/jwt/src/routes/v1/users/users.repository.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default class UsersRepository {
     return updatedUser ? updatedUser.toJSON() : null;
   }
 
-  public getAll(verified: boolean = true) {
+  public getAllVerifiedUsers(verified: boolean = true) {
```
Comment (Igrokqq) [inline generators/auth/templates/mongodb/jwt/src/routes/v1/users/users.repository.ts]: why do you need the  "verified" flag?

Comment (Igrokqq) [reply generators/auth/templates/mongodb/jwt/src/routes/v1/users/users.repository.ts]: you can simplify name of the method, just call him like getVerifiedUsers

File: generators/auth/templates/mongodb/oauth2/src/routes/v1/users/users.service.ts [hunk 2]
```diff
@@ -8,49 +8,49 @@ import UserDto from './dto/user.dto';
...
       verified,
     }).exec() as UserEntity;
 
-    return foundUser || null;
+    return foundUser.toJSON() || null;
   }
 
   async createIfDoesNotExist(user: UserDto): Promise<User | null> {
-    const foundUser = await this.usersRepository.findOne({
+    const foundUser = await this.usersModel.findOne({
```
Comment (Igrokqq) [inline generators/auth/templates/mongodb/oauth2/src/routes/v1/users/users.service.ts]: think about repository pattern

File: generators/auth/templates/mongodb/oauth2/src/interceptors/serialization.interceptor.ts [hunk 3]
```diff
@@ -15,7 +15,13 @@ export default class SerializeInterceptor implements NestInterceptor {
       map((args) => {
         const SerializeType = getSerializeType(context.getHandler());
         const entity = new SerializeType();
-        return Object.assign(entity, args);
+        let obj: any = {};
```
Comment (Igrokqq) [inline generators/auth/templates/mongodb/oauth2/src/interceptors/serialization.interceptor.ts]: refactor please

File: generators/auth/templates/mysql/jwt/src/interceptors/wrap-response.interceptor.ts [hunk 4]
```diff
@@ -13,6 +13,11 @@ export default class WrapResponseInterceptor implements NestInterceptor {
     return next.handle().pipe(
       map((...args) => {
         // if this an error response then return first object if no then second..
+        if (args[0].data) {
+          return {
+            data: args[0].data,
+          };
+        }
         return {
           data: args[0],
```
Comment (Igrokqq) [inline generators/auth/templates/mysql/jwt/src/interceptors/wrap-response.interceptor.ts]: (argument1) => {
  return {
    data: argument1.data || argument1
  }
}

File: generators/auth/templates/mongodb/oauth2/src/routes/v1/users/users.repository.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    return newUser.toJSON();
+  }
+
+  public async getByEmail(email: string): Promise<User | null> {
+    const user = await this.usersModel.findOne({
+      email,
+    }, { password: 0 }).exec();
+
+    return user ? user.toJSON() : null;
```
Comment (v-checha) [inline generators/auth/templates/mongodb/oauth2/src/routes/v1/users/users.repository.ts]: we must remove all toJSON methods
and continue work with mongoose instance

File: generators/auth/templates/mongodb/oauth2/src/routes/v1/users/users.repository.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  public async getVerifiedUserById(id: Types.ObjectId): Promise<User | null> {
+    const user = await this.usersModel.findOne({
```
Comment (v-checha) [inline generators/auth/templates/mongodb/oauth2/src/routes/v1/users/users.repository.ts]: better write
```
return this.usersModel.findOne({
      _id: id,
      verified: true,
    }, { password: 0 }).exec()
```

File: generators/auth/templates/mysql/jwt/src/routes/v1/users/users.repository.ts [hunk 7]
```diff
@@ -19,20 +19,55 @@ export default class UsersRepository {
...
-  public async getById(id: number, verified: boolean = true): Promise<UserEntity | null> {
+  public async getVerifiedUserByEmail(email: string) {
+    const user: UserEntity | undefined = await this.usersModel.findOne({
+      where: [{
+        email,
+        verified: true,
+      }],
+    });
+
+    return user || null;
```
Comment (v-checha) [inline generators/auth/templates/mysql/jwt/src/routes/v1/users/users.repository.ts]: the same 
```
return this.usersModel.findOne({
      where: [{
        email,
        verified: true,
      }],
    });
```

File: generators/auth/templates/mysql/oauth2/src/routes/v1/users/users.repository.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  public async getByEmail(email: string): Promise<UserEntity | null> {
+    const user: UserEntity | undefined = await this.usersModel.findOne({
+      where: [{
+        email
+      }],
+    });
+
+    return user || null;
```
Comment (v-checha) [inline generators/auth/templates/mysql/oauth2/src/routes/v1/users/users.repository.ts]: the same

File: generators/auth/templates/postgresql/jwt/src/routes/v1/users/users.repository.ts [hunk 9]
```diff
@@ -19,20 +19,51 @@ export default class UsersRepository {
...
   }
 
-  public async getById(id: number, verified: boolean = true): Promise<UserEntity | null> {
+  public async getVerifiedUserByEmail(email: string): Promise<UserEntity | null> {
+    const user: UserEntity | undefined = await this.usersModel.findOne({
+      email,
+      verified: true,
+    });
+
+    return user || null;
```
Comment (v-checha) [inline generators/auth/templates/postgresql/jwt/src/routes/v1/users/users.repository.ts]: the same

File: generators/jsonapi/templates/mongodb/jwt/src/routes/v1/auth/mailer-auth.controller.ts [hunk 10]
```diff
@@ -327,7 +327,10 @@ export default class AuthController {
       throw new NotFoundException();
     }
 
-    return {};
+    return ResponseUtils.success(
```
Comment (v-checha) [inline generators/jsonapi/templates/mongodb/jwt/src/routes/v1/auth/mailer-auth.controller.ts]: I guess we need remove ResponseUtils 
what do you thinks?

**Pull request merged**